src/glmfit.jl

```diff
@@ -233,10 +233,13 @@ function _fit!(m::AbstractGLM, verbose::Bool, maxIter::Integer, minStepFac::Real
         installbeta!(p, f)
         crit = (devold - dev)/dev
         verbose && println("$i: $dev, $crit")
-        if crit < convTol
+        if crit < convTol || dev == 0
             cvg = true
             break
         end
+        if !isfinite(crit)
+            error("convergence criterion is $crit") # maybe change ConvergenceException to allow for custom error
+        end
```

Comment:
Yeah, but it's not great if you need to automate the check to find out why convergence failed. Another possibly useful extension at some point would be to return the values of coefficients (this is e.g. what the gnm R package does and that's useful to find out which variables are problematic).